crates/re_chunk/src/chunk.rs

```diff
@@ -0,0 +1,478 @@
+    pub(crate) times: Vec<TimeInt>,
+
+    /// Is [`Self::times`] sorted?
+    ///
+    /// This is completely independent of [`Chunk::is_sorted`]: a timeline doesn't necessarily
+    /// follow the global [`RowId`]-based order, although it does in most cases (happy path).
+    pub(crate) is_sorted: bool,
+
+    /// The time range covered by [`Self::times`].
+    ///
+    /// Not necessarily contiguous! Just the min and max value found in [`Self::times`].
+    pub(crate) time_range: ResolvedTimeRange,
+}
+
+impl Default for ChunkTimeline {
+    #[inline]
+    fn default() -> Self {
+        Self {
+            times: Default::default(),
+            is_sorted: true,
+            time_range: ResolvedTimeRange::EMPTY,
+        }
+    }
+}
+
+#[cfg(test)] // do not ever use this outside internal testing, it's extremely slow and hackish
+impl PartialEq for Chunk {
+    #[inline]
+    fn eq(&self, rhs: &Self) -> bool {
+        let Self {
+            id: _, // we're comparing the contents
+            entity_path,
+            is_sorted,
+            row_ids,
+            timelines,
+            components,
+        } = self;
+
+        use itertools::Itertools as _;
+
+        *entity_path == rhs.entity_path
+            && *is_sorted == rhs.is_sorted
+            && *row_ids == rhs.row_ids
+            && *timelines == rhs.timelines
+            && components.keys().collect_vec() == rhs.components.keys().collect_vec()
+            && components.iter().all(|(component_name, list_array)| {
+                let Some(rhs_list_array) = rhs
+                    .components
+                    .get(component_name)
+                    .map(|list_array| &**list_array)
+                else {
+                    return false;
+                };
+
+                // `arrow2::compute::comparison` has very limited support for the different arrow
+                // types, so we just do our best here.
+                // This is just a testing/debugging tool.
+                if arrow2::compute::comparison::can_eq(list_array.data_type()) {
+                    arrow2::compute::comparison::eq(&**list_array, rhs_list_array)
+                        .values_iter()
+                        .all(|v| v)
+                } else {
+                    list_array.data_type() == rhs_list_array.data_type()
+                        && list_array.len() == rhs_list_array.len()
+                }
+            })
+    }
+}
+
+#[cfg(test)] // do not ever use this outside internal testing, it's extremely slow and hackish
+impl Eq for Chunk {}
+
+impl Chunk {
+    /// Creates a new [`Chunk`].
+    ///
+    /// This will fail if the passed in data is malformed in any way -- see [`Self::sanity_check`]
+    /// for details.
+    ///
+    /// Iff you know for sure whether the data is already appropriately sorted or not, specify `is_sorted`.
+    /// When left unspecified (`None`), it will be computed in O(n) time.
+    pub fn new(
```

Comment:
TODO in this PR or another: when creating a chunk of static data, there is no reason to keep anything but the last row (in sorted row-id order).

The backend will have to support multi-rows static chunks anyhow since clients can send anything, which both the query engine and compaction will know how to take care of, but it's a nice little optimization on the standard path.